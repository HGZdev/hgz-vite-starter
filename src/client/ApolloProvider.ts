import config from "../../config/config.ts";
import {makeApolloProvider} from "../../lib/apollo/ApolloClient.tsx";

const ApolloProvider = makeApolloProvider(config);
export default ApolloProvider;
