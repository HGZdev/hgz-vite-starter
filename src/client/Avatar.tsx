import styled from "styled-components";

const AvatarContainer = styled("div")`
  position: relative;
`;

const Avatar = styled("img")`
  filter: drop-shadow(0 0 10px rgba(0, 0, 0, 0.5));
`;

const FancyAvatar = () => {
  return (
    <AvatarContainer>
      <Avatar
        // alt="Favicon"
        src="/../assets/favicon.ico"
        // sx={{width: 250, height: 250}}
      />
    </AvatarContainer>
  );
};

export default FancyAvatar;
