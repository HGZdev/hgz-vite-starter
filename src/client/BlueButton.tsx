import styled from "styled-components";

export const BlueButton = styled.button`
  background-color: #0074d9;
  color: #fff;
  border: none;
  border-radius: 5px;
  padding: 10px 20px;
  font-size: 18px;
  cursor: pointer;
  transition: background-color 0.3s;

  &:hover {
    background-color: #0056b3;
  }
`;
