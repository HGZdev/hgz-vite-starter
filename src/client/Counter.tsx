import {Button, Typography, Container, CircularProgress} from "@mui/material";
import {useGetCounter, useIncrementCounter} from "../_server/queries";

const CounterContainer = () => {
  const {loading, error, data, refetch} = useGetCounter();
  const [incrementCounter] = useIncrementCounter();

  const handleIncrement = async () => {
    try {
      await incrementCounter();
      refetch();
    } catch (error) {
      const errorMessage = error as Error;
      console.error("Error incrementing counter:", errorMessage.message);
    }
  };

  return (
    <Container
      component="main"
      maxWidth="xs"
      style={{
        textAlign: "center",
        marginTop: "2rem",
        padding: "1rem",
        border: "2px solid #eeeeee",
        borderRadius: "8px",
      }}
    >
      <Typography variant="h4">Counter</Typography>
      {loading ? (
        <CircularProgress />
      ) : error ? (
        <Typography color="error">Error: {error.message}</Typography>
      ) : (
        <>
          <Typography variant="h5" gutterBottom>
            Current Count: {data?.getCounter.value ?? 0}
          </Typography>
          <Button
            variant="contained"
            color="primary"
            onClick={handleIncrement}
            aria-label="Increment the counter"
          >
            Increment
          </Button>
        </>
      )}
    </Container>
  );
};

export default CounterContainer;
