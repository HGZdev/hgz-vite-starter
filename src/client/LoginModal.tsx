import React, {useState} from "react";
import {useLogin} from "../_server/queries";
import styled from "styled-components";

const ModalBackground = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.8);
`;

const ModalContent = styled.div`
  background-color: white;
  padding: 2rem;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
  width: 100%;
  max-width: 24rem;
`;

const FormGroup = styled.div`
  margin-bottom: 1.5rem;
`;

const Label = styled.label`
  display: block;
  margin-bottom: 0.5rem;
  font-weight: bold;
  color: #4a5568;
`;

const Input = styled.input`
  width: calc(100% - 1rem);
  padding: 0.5rem;
  border: 1px solid #e2e8f0;
  border-radius: 0.25rem;
`;

const ErrorMessage = styled.div`
  color: #ef4444;
  margin-bottom: 1rem;
`;

const SubmitButton = styled.button`
  width: 100%;
  padding: 1rem;
  background-color: #2563eb;
  color: white;
  border-radius: 0.25rem;
  transition: background-color 0.3s ease;

  &:hover {
    background-color: #1d4ed8;
  }
`;

interface LoginModalProps {
  onClose: () => void;
}

const LoginModal: React.FC<LoginModalProps> = ({onClose}) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState<string | null>(null);

  const [login] = useLogin();

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    setErrorMessage(null);
    try {
      const {data} = await login({email, password});
      const token = data?.login?.token;

      if (token) {
        // setCookie("token", token, 1);
        onClose();
      }
    } catch (err) {
      if (err instanceof Error) setErrorMessage(err.message); // Error handling
    }
  };

  return (
    <ModalBackground>
      <ModalContent>
        <form onSubmit={handleSubmit}>
          <FormGroup>
            <Label htmlFor="email">Email:</Label>
            <Input
              type="email"
              id="email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              required
            />
          </FormGroup>
          <FormGroup>
            <Label htmlFor="password">Password:</Label>
            <Input
              type="password"
              id="password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
            />
          </FormGroup>
          {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
          <SubmitButton type="submit">Login</SubmitButton>
        </form>
      </ModalContent>
    </ModalBackground>
  );
};

export default LoginModal;
