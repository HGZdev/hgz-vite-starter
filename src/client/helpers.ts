import {serialize} from "cookie";

export const setCookie = (
  name: string,
  value: string,
  expiryHours: number
): string => {
  const expires = new Date();
  expires.setTime(expires.getTime() + expiryHours * 60 * 60 * 1000);

  const cookie = serialize(name, value, {
    expires,
    path: "/",
  });

  document.cookie = cookie;
  return cookie;
};

export const destroyCookie = (name: string): string => {
  const expires = new Date();
  expires.setTime(expires.getTime() - 1);

  const cookie = serialize(name, "", {
    expires,
    path: "/",
  });

  document.cookie = cookie;
  return cookie;
};
