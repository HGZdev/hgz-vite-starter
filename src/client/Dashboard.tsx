import {Button, Typography, Container} from "@mui/material";
import {useGetUserMe, useLogout} from "../_server/queries";
import Counter from "./Counter";
import styled from "styled-components";

const FlexContainer = styled(Container)`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100vh;
`;

const Title = styled(Typography)`
  margin-bottom: 20px;
`;

const LogoutButton = styled(Button)`
  align-self: flex-end;
  margin-top: 20px;
`;

const Dashboard = () => {
  const {loading, error, data} = useGetUserMe();
  const [logout] = useLogout();

  const handleLogout = async () => {
    await logout();
  };

  if (!data && loading)
    return (
      <FlexContainer aria-busy="true" aria-live="polite">
        Loading...
      </FlexContainer>
    );
  if (error)
    return (
      <FlexContainer aria-live="assertive">
        Error: {error.message}
      </FlexContainer>
    );

  return (
    <FlexContainer>
      <LogoutButton
        variant="contained"
        color="primary"
        onClick={handleLogout}
        aria-label="Logout Button"
      >
        Logout
      </LogoutButton>
      <div>
        <Title variant="h4">Welcome to the Dashboard!</Title>
        <Counter />
      </div>
    </FlexContainer>
  );
};

export default Dashboard;
