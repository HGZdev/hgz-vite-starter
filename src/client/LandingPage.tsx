import React, {useState} from "react";
import styled from "styled-components";
import {useGetUserMe} from "../_server/queries";
import LoginModal from "./LoginModal";
import Dashboard from "./Dashboard";

const LandingPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
`;

const SignupButton = styled.button`
  padding: 10px 20px;
  font-size: 16px;
  background-color: #007bff;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
`;

const LoginButton = styled.button`
  margin-top: 10px;
  padding: 10px 20px;
  font-size: 16px;
  background-color: #28a745;
  color: #fff;
  border: none;
  border-radius: 4px;
  cursor: pointer;
`;

const LandingPage: React.FC = () => {
  const [showLoginModal, setShowLoginModal] = useState(false);
  const {data, loading, error} = useGetUserMe();

  if (!data && loading) return <p>Loading...</p>;
  if (error) return <p>Error: {error.message}</p>;

  const getUserMe = data?.getUserMe;

  if (showLoginModal)
    return <LoginModal onClose={() => setShowLoginModal(false)} />;

  return (
    <LandingPageContainer>
      {!getUserMe ? (
        <>
          <SignupButton
            aria-label="Sign up"
            onClick={() => (window.location.href = "/registration")}
          >
            Sign up
          </SignupButton>
          <LoginButton
            aria-label="Login"
            onClick={() => setShowLoginModal(true)}
          >
            Log in
          </LoginButton>
        </>
      ) : (
        <Dashboard />
      )}
    </LandingPageContainer>
  );
};

export default LandingPage;
