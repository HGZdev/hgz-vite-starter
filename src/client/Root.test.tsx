import {render, screen, fireEvent} from "@testing-library/react";
import {describe, it, expect} from "vitest";
import {MockedProvider} from "@apollo/client/testing";
import App from "./Root";
import {getCounterGQL, incrementCounterGQL} from "../_server/queries";

const mockedValue = 5;

// Adjusting GraphQL mocks to include the `id` field and the value wrapped as `{ value: number }`
const mocks = [
  {
    request: {
      query: getCounterGQL,
    },
    result: {
      data: {counter: {id: "1", value: mockedValue}},
    },
  },
  {
    request: {
      query: incrementCounterGQL,
    },
    result: {
      data: {incrementCounter: {id: "1", value: mockedValue + 1}}, // Incremented value is also wrapped in an object
    },
  },
  {
    request: {
      query: getCounterGQL,
    },
    result: {
      data: {counter: {id: "1", value: mockedValue + 1}}, // Next fetch also provides incremented value
    },
  },
];

describe("App Component", () => {
  it("renders without crashing", async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>
    );

    // screen.debug();

    // Check if "Loading..." text appears first
    const loadingText = screen.queryByText(/Loading.../i);
    expect(loadingText).toBeTruthy();

    // Wait for the counter to display
    const counterDisplay = await screen.findByText(
      new RegExp(`Counter: ${mockedValue}`)
    );
    expect(counterDisplay).toBeTruthy();

    // Click the increment button
    const incrementButton = screen.getByText(/Increment/i);
    fireEvent.click(incrementButton);

    // Wait for the updated counter value to display
    const updatedCounter = await screen.findByText(
      new RegExp(`Counter: ${mockedValue + 1}`)
    );
    expect(updatedCounter).toBeTruthy();
  });
});
