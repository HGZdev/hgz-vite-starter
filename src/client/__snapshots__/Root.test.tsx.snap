// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root Component Tests > User is logged-out > renders landing page when path is "/" 1`] = `
<div>
  <div
    class="sc-hKgKIp jbUdmH"
    data-testid="LandingPage"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-zxyrzw-MuiAvatar-root"
    >
      <img
        alt="Favicon"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/../assets/favicon.ico"
      />
    </div>
    <div>
      Welcome!
    </div>
    <div
      class="sc-eCstZk hQghqi"
    >
      <button
        aria-label="Register"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-jSguLX css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Register
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Login"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-gKsecS css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
